[MAP-52] Kakao API와 연동해서 장소 정보를 검색하는 기능을 구현했습니다.

hwankim123 opened the pull request:
⭐️ 좋은 코드 리뷰는 친절한 PR로 부터 나옵니다! 모두들 꼼꼼히 작성해주세요. ⭐️

## 업무 배경
> 업무 이유, 목적, 이슈 등 업무의 배경을 간단하게 1줄 요약 합니다.

* 여기에 내용 기재

## 개발 내용
> 수정 or 추가, 개발된 내용에 대해서 1줄로 간단히 작성합니다.

* Kakao API와 연동해서 장소 정보를 검색하는 기능을 구현했습니다.

## 집중적인 코드리뷰
> 좀 더 상세하게 리뷰 받고 싶은 부분을 기재해주세요.

* 범석님이 구현하신 "성공한 Response를 위한 로직"을 적용했습니다. 제대로 적용했나 확인해주시면 감사하겠습니다.
* api/v1/map/keyword : 장소에 대한 키워드를 검색하면 keyword로 장소를 검색하는 Kakao API에 요청을 보냅니다. 해당 키워드와 관련있는 장소 정보를 응답합니다.
* api/v1/map/coordinate : 위도, 경도에 대한 요청이 들어오면 좌표를 통해 장소를 검색하는 Kakao API에 요청을 보냅니다. 해당 좌표의 주소 정보 등을 응답합니다.

## 리마인더
* 본인의 로컬에서 정상 동작하는지 확인해주세요.
* 최신 브랜치를 Pull 받고 PR을 요청했는지 확인해주세요.
* **API**가 추가되었을 경우 테스트를 하고 PR을 올려주세요.
* **Conflict**가 났을 때, UI상에서 해결하지 말고, 본인 local에서 해결해주세요.
* **commit 메시지** 제대로 작성해주세요.

hwankim123 committed:
[MAP-52] chore: Jackson 라이브러리 추가

hwankim123 committed:
[MAP-52] feat: Kakao API를 통해 장소 정보 조회하는 기능 구현

- Naver API는 비즈니스 요구사항과 맞지 않아서 Kakao API로 대체
- 장소에 대한 키워드 -> 장소 정보(위경도, 주소 등) 조회
- 위경도 -> 장소 정보(위경도, 주소 등) 조회
- request가 성공한 경우에 DataResponse로 DTO를 감싸서 응답

myeongju-kim reviewed (approved):
확인했습니다!
현재 액션 업데이트가 안되어 있어서 테스트에는 실패합니다 ㅠㅠ 머지하면 성공할겁니다.

qjatjr29 commented on src/main/java/com/mapnote/mapnoteserver/domain/map/dto/KakaoApiResponse.java:
```diff
@@ -0,0 +1,64 @@
+
+  @Getter
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
```
@JsonIgnoreProperties 어노테이션을 사용해서 역직렬화할 때 예외를 던지지 않도록 했군요!

qjatjr29 commented on src/main/java/com/mapnote/mapnoteserver/domain/map/dto/KakaoApiResponse.java:
```diff
@@ -0,0 +1,64 @@
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class Addresses{
+    private RoadAddress road_address;
```
여기서 RoadAddress와 Address를 나눈 이유가 있을까요?
```java
public static class Address {
  private String addressName;
  private String roadAddressName;
} 
```
이런식으로 사용해도 될거 같은데 이유가 궁금합니다!

hwankim123 replied:
KakaoApiResponse DTO는 Kakao API response JSON을 파싱하는데 사용되는 클래스입니다.
현재 사용중인 Jackson은 JSON 형식과 맞게 클래스를 나누어야 한다고 알고 있고,
원래 Address와 RoadAddress에 들어가는 JSON 데이터들이 많은데 비스니스 로직상 사용할만한 데이터가 저 address_name 하나밖에 없기 때문에 클래스에 필드가 하나밖에 없는 형태가 된 것입니다.

qjatjr29 replied:
Kakao API response JSON형식이 Address 와 RoadAddress를 구분한 형식이라는 걸로 이해하면 되나요?

hwankim123 replied:
넵 맞습니다

qjatjr29 commented on src/main/java/com/mapnote/mapnoteserver/domain/map/service/openapi/KakaoApi.java:
```diff
@@ -0,0 +1,79 @@
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Service
+public class KakaoApi {
```
네이밍에 Service가 여기에는 없는 이유가 있을 까요?

hwankim123 replied:
딱히 이유는 없었습니다. 네이밍을 통일성있게 작성하는게 더 좋을 것 같네요!

qjatjr29 reviewed (commented):
LGTM 👍

hwankim123 reviewed (commented):
리뷰남긴거 확인했습니다!

hwankim123 committed:
[MAP-52] refactor: KakaoApiService 코드 리팩토링

- 클래스명 변경 : KakaoApi -> KakaoApiService
- private static final 필드들의 필드 이름 수정

hwankim123 committed:
[MAP-52] feat: 장소 검색 기능에 Swagger 적용

- Swagger는 GET 요청을 request body로 보내지 못하고, Kakao API도 GET요청을 query pamarmeter로 받기 때문에 장소 검색 요청 데이터를 query parameter로 넘기도록 수정
- SwaggerConfig, Controller, DTO에 Swagger 설정 추가

hwankim123 committed:
[MAP-52] refactor: 불필요한 주석 삭제

hwankim123 committed:
Merge branch 'test' into feature/MAP-52
